### ISSUE TITLE
Webpack 3 bundle size is larger than Webpack 2.6.1

### ISSUE THREAD

**Issue by alexilyaev** [webpack/webpack#5126]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
The build passes, but fails to concatenate, bundle size is bigger than with 2.6.1
```
ModuleConcatenation (inner): module is used with non-harmony imports from multi react-hot-loader/patch ./app.js
ModuleConcatenation: Cannot concat with ../node_modules/react-dom/index.js
ModuleConcatenation: Cannot concat with ../node_modules/react/react.js
ModuleConcatenation: Cannot concat with ./assets/styles/main.scss
ModuleConcatenation: Cannot concat with ../node_modules/cssrecipes-defaults/lib/box-sizing.css
ModuleConcatenation: Cannot concat with ../node_modules/cssrecipes-defaults/lib/document-remove-margin-padding.css
ModuleConcatenation: Cannot concat with ../node_modules/cssrecipes-defaults/lib/hidden.css
ModuleConcatenation: Cannot concat with ../node_modules/normalize.css/normalize.css
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Tested on:
https://github.com/alexilyaev/react-es6-starter

See webpack3 branch:
alexilyaev/react-es6-starter#10

Run `npm run build`

**What is the expected behavior?**
I would expect the bundle size to at least stay the same...  
And if I can figure out how to make `ModuleConcatenationPlugin` work, to be smaller.

Bundle size results:
Webpack 2.6.1: `268KB` (67.3KB Gzip)
Webpack 3.0.0: `313KB` (78.3KB Gzip)
Webpack 3.0.0: `317KB` (78.6KB Gzip) (without `ModuleConcatenationPlugin`)

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
macOS 10.12.5
Node 6.10.3
Everything else in package.json

**Comment by aseem2625** [comment]
@alexilyaev 
yes there is a bug. But apparently, the bug seems be in html plugin which has unupdated dependencies. Check [this](https://github.com/webpack/webpack/issues/5101#issuecomment-309983784) to fix your repo..

**Comment by hyy1115** [comment]
I also encountered this phenomenon, but I did not use html-webpack-plugin

**Comment by aseem2625** [comment]
That's strange, @hyy1115 Can u plz share your repo to reproduce issue.

**Comment by hyy1115** [comment]
@aseem2625 Need to create an issue, or comments on this issue behind

**Comment by alexilyaev** [comment]
@hyy1115 If it's the same problem, you can comment here.

I'll try to remove html-webpack-plugin and see if it's related.

**Comment by alexilyaev** [comment]
I've tried to remove html-webpack-plugin completely, go the same results:  
https://github.com/alexilyaev/react-es6-starter/tree/webpack3-no-html-plugin

**Comment by hyy1115** [comment]
**Do you want to request a feature or report a bug?**
bug

**What is the current behavior?**
The compiled js files larger than webpack2.6.1。

**If the current behavior is a bug, please provide the steps to reproduce.**
[https://github.com/hyy1115/react-redux-webpack2](url)

```
npm run build-mac

```

```
Version: webpack 2.6.1
Time: 19446ms
               Asset      Size  Chunks                    Chunk Names
           vendor.js    273 kB       3  [emitted]  [big]  vendor

```

Now I put the webpack upgrading to 3.0.0, and add new webpack. Optimize the ModuleConcatenationPlugin (), and then execute the build
```
Version: webpack 3.0.0
Time: 23375ms
               Asset      Size  Chunks                    Chunk Names
           vendor.js    310 kB       3  [emitted]  [big]  vendor
```

**What is the expected behavior?**
I don't want to vendor. Js increase a few KB, and compile time should not be longer

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
mac: 10.12.5
node: 6.10.3
chrome:  59.0.3071.109

**Comment by sokra** [comment]
It's a bug in the ExtractTextPlugin, which is not yet webpack 3 compatible.

**Comment by alexilyaev** [comment]
In my case I"m not using ExtractTextPlugin.

**Comment by SidKwok** [comment]
Did you guys use react-hot-loader3? I found `sourcemapped-stacktrace.js` was packed in bundle when I use webpack3.0, but webapck2.x is ok. Same config.

**Comment by alexilyaev** [comment]
@SidKwok Looks like you're onto something...

The offending line:
```
  entry: {
    app: [
      'react-hot-loader/patch',
      './app.js'
    ]
  },
```

Tried putting a `console.log` and a `throw` in some files under `node_modules/react-hot-loader/` but I don't see anything when running the webpack build, so not sure how to debug.

**Comment by SidKwok** [comment]
@alexilyaev You can use [webpack-bundle-analyzer](https://github.com/th0r/webpack-bundle-analyzer) to see that `sourcemapped-stacktrace.js` was packed in bundle when using webpack 3.0

I can't really tell, but I have a strong feeling that it was something wrong with hot-reloading in webpack 3.0

**Comment by Faradey27** [comment]
I have the same issue and bigger bundle related to react-hot-loader,

So, i checked that in my webpack.prod.config.js - there is no anything related to hot loader, the only one place where it used

```import { AppContainer as HotEnabler } from 'react-hot-loader';``` in index.js file(root of project)

And what i could see with webpack 2.2: everything is fine, bundle size 285 kb,
With webpack 3(just changed version): 323kb,

source-map-explorer telling me that with webpack 3, i have in the bundle

react-hot-loader - 4 kb
sourcemapped-stacktrace - 20 kb
redbox-react - 10kb

and it also telling me, that i do not have such libs in prod bundle with webpack 2

Question, why after migration to webpack 3, react-hot-reload appeared in bundle? with webpack 2.x everything is fine

@SidKwok @gaearon @aseem2625

**Comment by SidKwok** [comment]
Have you guys tried 3.1.0? I think it solve the issue.

**Comment by Faradey27** [comment]
@SidKwok looks like yes, thank you

do you know what was the problem?

**Comment by alexilyaev** [comment]
I can confirm 3.1.0 fixed the issue, I am now getting `264kb` (66.7Gzip) compared to `268KB` (67.3KB Gzip) using 2.6.1.

Now can someone explain the bailout?

```
ModuleConcatenation (inner): module is used with non-harmony imports from multi react-hot-loader/patch ./app.js
```

See the first comment.

**Comment by loris** [comment]
Looks better, but I'm still confused, ModuleConcatenation is supposed to reduce your bundle size no? (there is a 50% size reduction linked in the official release blog post:  https://twitter.com/tizmagik/status/876128847682523138). Interested to hear some explanation around the bailouts, and why some people are seeing dramatic size reduction and others don't.
